docs/proposals/30-apiserver-proxy.md

```diff
@@ -0,0 +1,317 @@
+
+```yaml
+{{- if eq .Values.vpn.enabled true -}}
+apiVersion: networking.istio.io/v1beta1
+kind: Gateway
+metadata:
+  name: reversed-vpn-auth-server
+  namespace: {{ .Release.Namespace }}
+spec:
+  selector:
+{{ .Values.labels | toYaml | indent 4 }}
+  servers:
+  - hosts:
+    - reversed-vpn-auth-server.garden.svc.cluster.local
+    port:
+      name: tls-tunnel
+      number: 8134
+      protocol: HTTP
+{{- end }}
+```
+### Implementation Steps
+
+1. Apply the Istio Gateway and VirtualService configurations.
+2. Deploy the EnvoyFilter for custom header processing.
+3. Update any relevant Istio authorization policies to allow the new traffic flow.
+4. Update the `apiserver-proxy` code or configuration to use HTTP CONNECT and include the custom header.
+5. Deploy the updated `apiserver-proxy` configuration.
+
+## Phased Rollout
+
+The implementation will be controlled by two distinct feature gates that handle different aspects of the solution. Each feature gate can be disabled per shoot to allow testing of the old implementation via E2E tests.
+
+### Feature Gate 1: `APIServerSecureRouting`
+Controls the enablement of the new routing implementation using HTTP CONNECT and custom headers.
+
+#### Alpha
+- **Default:** Disabled
+- **Actions:**
+    - Introduces new port for HTTP CONNECT
+    - Deploys new custom header processing
+    - Configures mTLS for new routing path
+    - Enables E2E testing of the feature
+    - Implementation of all unit and integration tests
+    - Documentation completion
+
+#### Beta
+- **Default:** Enabled
+- New shoots automatically use new implementation
+- Existing shoots continue using old implementation
+
+#### GA
+- **Default:** Always enabled
+- All shoots use new implementation
+- Existing shoots automatically migrated during reconciliation
+
+### Feature Gate 2: `APIServerLegacyPortDisable`
+Controls the disabling of the legacy port. Can only progress once `APIServerSecureRouting` is GA.
+
+#### Alpha
+- **Default:** Disabled
+- **Prerequisites:**
+    - Feature gate `APIServerSecureRouting` is in GA state
+    - All monitored shoots successfully using the new implementation
+- **Actions:**
+    - Marks port as deprecated
+    - Implementation of E2E tests with legacy port disabled
+
+#### Beta
+- **Default:** Enabled
+- Automatically closes legacy port for new shoots
+- Existing shoots prompted for migration during reconciliation
+
+#### GA
+- **Default:** Always enabled
+- Legacy port completely disabled
+- All configuration for legacy port removed
+
+### Testing Strategy
+- Feature gates can be switched off per shoot to allow testing of the old implementation
+- E2E tests will cover both new and old implementations
+- All testing functionality is implemented and validated in alpha phase
+
+## Alternatives
+
+- Implement stricter filtering of proxy protocol headers at the istio-ingress gateway level
+- Use a custom LoadBalancer solution that is transparent or aware of the Gardener architecture
```

Comment:
Not all infrastructures provide transparent load balancers. Requiring a transparent load balancer for Gardener is no solution. However, it is fair to have this as a prerequisite for the ACL extension.